emulated/emulated.go

```diff
@@ -0,0 +1,366 @@
+package emulated
+
+import (
+	pb "buf.build/gen/go/meshtastic/protobufs/protocolbuffers/go/meshtastic"
+	"context"
+	"fmt"
+	"github.com/charmbracelet/log"
+	"github.com/crypto-smoke/meshtastic-go"
+	"github.com/crypto-smoke/meshtastic-go/mqtt"
+	"github.com/crypto-smoke/meshtastic-go/radio"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/protobuf/proto"
+	"sync"
+	"time"
+)
+
+// Config is the configuration for the emulated Radio.
+type Config struct {
+	// Dependencies
+	MQTTClient *mqtt.Client
+
+	// Node configuration
+	// NodeID is the ID of the node.
+	NodeID meshtastic.NodeID
+	// LongName is the long name of the node.
+	LongName string
+	// ShortName is the short name of the node.
+	ShortName string
+	// Channels is the set of channels the radio will listen and transmit on.
+	// The first channel in the set is considered the primary channel and is used for broadcasting NodeInfo and Position
+	Channels *pb.ChannelSet
+	// BroadcastInterval is the interval at which the radio will broadcast NodeInfo and Position messages.
+	// Defaults to 5 minutes.
+	BroadcastInterval time.Duration
```

Comment:
I would like to see separate broadcast intervals for nodeinfo and position, with a setting of zero disabling them. I don't think we need to be sending position packets periodically in an example app that many will likely use to learn this package.

Reply:
Addressed in 5a06b85